vertx-web/src/main/java/io/vertx/ext/web/impl/PrefixedRouterImpl.java

```diff
@@ -0,0 +1,134 @@
+package io.vertx.ext.web.impl;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.Route;
+
+import java.util.regex.Pattern;
+
+/**
+ * A router which adds a prefix to every route added to it.
+ * It's useful in place where a router is accepted and is used as a sub-router.
+ * A sub-router can't be used with regular expressions or named path segments.
+ *
+ * <pre>
+ *     Router prefixedRouter = Router.prefixedRouter(vertx, "/api/:name/events");
+ *     SockJSHandler sockHandler = SockJSHandler.create(vertx, prefixedRouter).bridge(options)
+ *
+ *     Router httpRouter = Router.create(vertx)
+ *     httpRouter.route("/*").handler(sockHandler)
+ * </pre>
+ *
+ * @author jansorg
+ */
+public class PrefixedRouterImpl extends RouterImpl {
```

Comment:
Looking at the prefixed router I am wondering, won’t this solve the issues we have with sub routers? Could we attempt to replace the current sub routers with this?